android/build.gradle

```diff
@@ -98,6 +98,86 @@ if (androidSdkInstalled) {
         })
         provided project(':conscrypt-android-stub')
     }
+
+    task javadocs(type: Javadoc) {
+        source = android.sourceSets.main.java.srcDirs
+        classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
+        // TODO(nmittler): Fix the javadoc errors.
+        failOnError false
+        options {
+            encoding = 'UTF-8'
+            links "http://docs.oracle.com/javase/7/docs/api/"
+        }
+    }
+
+    task javadocsJar(type: Jar, dependsOn: javadocs) {
+        classifier = 'javadoc'
+        from javadocs.destinationDir
+    }
+
+    task sourcesJar(type: Jar) {
+        classifier = 'sources'
+        from android.sourceSets.main.java.srcDirs
+    }
+
+    artifacts {
+        archives sourcesJar
+        archives javadocsJar
```

Comment:
It appears to work: https://oss.sonatype.org/content/repositories/snapshots/org/conscrypt/conscrypt-android/1.1.0-SNAPSHOT/. I think the problem before was the lack of a configuration for `uploadArchives`